Assets/Scripts/Models/Furniture.cs

```diff
@@ -366,6 +357,28 @@ protected bool DEFAULT__IsValidPosition(Tile t)
         return true;
     }
 
+
+    public bool HasPowerCheck()
+    {
+
+        if (powerValue > 0 && isPowerGenerator == false)
```

Comment:
Maybe easier would be to change `powerValue` and `isPowerGenerator` to something like:
`powerConsumption`
and just play with positive/negative;

for example 
if negative - that this is power supply
if positive - that is consumer

That you do not need: `<PowerGenerator supply="5"/>`